dotnet/src/Agents/Abstractions/AggregatorChannel.cs

```diff
@@ -49,6 +50,36 @@ protected internal override IAsyncEnumerable<ChatMessageContent> GetHistoryAsync
         }
     }
 
+    /// <inheritdoc/>
+    protected internal override async IAsyncEnumerable<StreamingChatMessageContent> InvokeStreamingAsync(AggregatorAgent agent, IList<ChatMessageContent> messages, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
It looks like `messages` parameter is used as output only. Since we are returning each message as `IAsyncEnumerable`, do we need additional `messages` parameter? In `Kernel`, `messages` are also used as input parameter, so it's possible to continue the conversation based on previous messages. Here, it looks like `messages` parameter is used only as output, but we already return `IAsyncEnumerable<StreamingChatMessageContent>`, so I'm not sure we need both, unless I'm missing something. 

But even if this case is required, maybe we should consider marking this parameter with `out` keyword to clearly show that this parameter is used as output only. Otherwise, no matter what we pass to this parameter is input, it won't have impact, which may be confusing when using this method.

Reply:
Enumerator methods don't support 'out' keyword.  In any case, the contract here is to append the fully-formed message to an existing collection of messages in order to maintain the history...irregardless of making the streamed content available for display.